Source/WebCore/layout/formattingContexts/inline/InlineContentConstrainer.cpp

```diff
@@ -174,19 +174,24 @@ void InlineContentConstrainer::updateCachedWidths()
         auto isWordSeparator = false;
         if (auto* textItem = dynamicDowncast<InlineTextItem>(item))
             isWordSeparator = textItem->isWordSeparator();
-
-        m_inlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, false) +  (isWordSeparator ? item.style().wordSpacing() : 0.0f);
-        m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_inlineItemWidths[i]);
-        m_firstLineStyleInlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, true) + (isWordSeparator ? item.firstLineStyle().wordSpacing() : 0.0f);
-        m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_firstLineStyleInlineItemWidths[i]);
+        // Opaque items are ignored by inline layout. Skip over this item by setting its width to 0.
+        if (item.isOpaque()) {
+            m_inlineItemWidths[i] = 0;
+            m_firstLineStyleInlineItemWidths[i] = 0;
+        } else {
+            m_inlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, false) +  (isWordSeparator ? item.style().wordSpacing() : 0.0f);
+            m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_inlineItemWidths[i]);
+            m_firstLineStyleInlineItemWidths[i] = m_inlineFormattingContext.formattingUtils().inlineItemWidth(item, 0, true) + (isWordSeparator ? item.firstLineStyle().wordSpacing() : 0.0f);
+            m_inlineItemWidthsMax = std::max(m_inlineItemWidthsMax, m_firstLineStyleInlineItemWidths[i]);
+        }
     }
     m_hasValidInlineItemWidthCache = true;
 }
 
 void InlineContentConstrainer::checkCanConstrainInlineItems()
 {
     for (auto& inlineItem : m_inlineItemList) {
-        if (cannotConstrainInlineItem(inlineItem)) {
+        if (!inlineItem.isOpaque() && cannotConstrainInlineItem(inlineItem)) {
```

Comment:
I'd put the `isOpaque()` check in `cannotConstrainInlineItem` to future proof it (in case you start calling it from some other function)